### PR TITLE
Refine lifecycle module generator formatting

### PR DESCRIPTION
**Pull request by leynos**
## Summary
- refactor the lifecycle test generator to build modules from a shared body map while conditionally injecting helper imports, decorators, and rejecting unknown phases
- ensure auto-fail modules omit subprocess helpers and support multi-line decorators through explicit formatting logic
- extend the unit suite with negative phase coverage and decorator assertions to lock in the regenerated module layout

## Testing
- make check-fmt
- make typecheck
- make lint
- make test

------
https://chatgpt.com/codex/tasks/task_e_68d81ec556c88322aadbee30e293153c

## Summary by Sourcery

Refine the pytest lifecycle module generator by consolidating its templating logic, improving decorator and import handling, renaming the failure flag, and adding explicit phase validation

New Features:
- Add support for pure auto-fail modules that omit subprocess helpers via an explicit expect_auto_fail parameter
- Validate lifecycle phases by raising a ValueError for unknown phases
- Preserve and dedent multi-line decorators when injecting them into generated test modules

Enhancements:
- Refactor the lifecycle test generator to build modules from a shared body map with conditional header and shim code insertion
- Rename the failure flag from should_fail to expect_auto_fail and expand the generate_lifecycle_test_module docstring for clarity

Tests:
- Add unit tests for negative phase validation, pure auto-fail modules, multi-line decorators, and unknown phase errors
- Update existing tests to use the expect_auto_fail parameter

## Summary by CodeRabbit

- New Features
  - Tests now generate modules with more flexible assembly and improved decorator formatting (preserves whitespace and multi-line decorators).
- Bug Fixes
  - Validation added with a clearer error when an unknown phase is provided.
- Refactor
  - Public parameter renamed to expect_auto_fail; all call sites updated to match.
- Tests
  - Expanded coverage for auto-fail scenarios, omission of helpers, multiline decorator handling, and invalid phases.
- Documentation
  - Updated inline descriptions to reflect the renamed parameter and adjusted behaviors.

### CODE REVIEW

File: cmd_mox/unittests/test_pytest_plugin_module_utils_unit.py [hunk 1]
```diff
@@ -83,3 +98,45 @@ def test_case(cmd_mox):
...
+
+
+def test_generate_module_rejects_unknown_phases() -> None:
+    """An invalid phase should raise an explicit error for callers."""
+    with pytest.raises(ValueError, match="Unknown phase"):
+        plugin_utils.generate_lifecycle_test_module(
+            decorator="",
+            expected_phase=t.cast("plugin_utils.PhaseLiteral", "INVALID"),
+            expect_auto_fail=False,
+        )
```
[the remark this answers is not in the document]
Comment (leynos) [reply cmd_mox/unittests/test_pytest_plugin_module_utils_unit.py:142]: @coderabbitai typing.cast takes a string or a type object if `from __future__ import annotations` is used. Using a string means that we do not have to import the type at runtime if it is not used at runtime anywhere else in the file.
